### PR TITLE
Rename Cotx-Networks.mdx and Update light-hoptspots.mdx

### PR DESCRIPTION
**Pull request by cotxliang**
# Cotx X3 LoRaWAN Gateway Setup

This guide provides instructions on how to configure a Cotx X3 LoRaWAN gateway to work with the Helium network.

:::warning

Cotx X3 gateway will only mine HNT through packet transfer, and is considered a non-PoC Light Hotspot. The ability to mine HNT with anything other than [Approved Hotspots](https://helium.com/mine) is not currently supported. Please join the [#gateway-development](https://discord.gg/helium) channel in the Helium Discord Server for the latest updates.

:::

## Prerequisites

Prior to configuring your Cotx X3 to work with the Helium network, please confirm that it is connected to the Internet and updated to the latest firmware. 

- Connect to a Wi-Fi or Ethernet network with Internet access.
- Use SSH. Your username is `admin` and the default password is `123xxxxxx`, please contact the salesperson for the password.
- Once logged in, please change your password.

## Configure LoRa Frequency Plan

To be able to send and receive LoRa data, your gateway must be configured to the frequencies specified for your region. You may be able to use the predefined frequency in the Cotx X3 or some cases may need to manually set them per the frequencies used by Helium in your region.

1. Under Radio Settings, select your region in the [Frequency Plan](https://docs.helium.com/lorawan-on-helium/frequency-plans) drop down and, if applicable, the Frequency Sub Band. For example, in the US, select US915 for the frequency plan and 2 for the sub band.
2. For some regions, it my be necessary to manually populate Populate the Customized Channel Settings based on the Helium frequencies for your region.

Refer to the [Helium sample LoRa packet forwarder global.conf files](https://github.com/helium/sx1302_hal/tree/helium/hotspot/packet_forwarder) for your region as a guide to the values to use. You can check the values used by the pre-defined frequency plans by viewing the global_conf.json file: `cat /root/helium/lora_net/lora_pkt_fwd/global_conf.json`

## Install Helium Gateway (gateway-rs) on the Cotx X3

If your supported LoRa gateway did not come with helium-gateway pre-installed, manual installation requires you to:

1. Configure the packet forwarder on the gateway to forward to the helium-gateway application. This varies per gateway but the goal is to set the packet forwarder to forward to the (default) configured helium-gateway on `127.0.0.1` at udp port `1680`
2. Connect to the Cotx X3 over SSH. 
3. `scp` a downloaded release package for the supported platform to the gateway. Copy the Helium Gateway package to the Cotx X3. Run this command from on the computer within the directory containing the package `scp helium-gateway-<version>-<platform>.deb <gateway>:/tmp/`.
4. Install the Helium Gateway package `dpkg --install /tmp/helium-gateway-<version>-<platform>.deb`.
5. The default region of the gateway is `US915`, if your region is different you can set the right one in `/etc/helium_gateway/settings.toml`. Just add the following line : `region = "<region>"` . Possible values are : `US915| EU868 | EU433 | CN470 | CN779 | AU915 | AS923 | KR920 | IN865`.
6. Backup your key file `/etc/helium_gateway/keypair.bin` and keep in a safe place. This will be needed to recover this gateway especially if it is onboarded to the blockchain.

Helium Gateway is now installed and should be running. Use the command `helium_gateway key info` to view the public key address and animal name of your gateway. 

## Verify

Your Cotx X3 gateway should now be setup and forwarding LoRaWAN data traffic to the Helium Network.

### Through the System Logs

Use the Cotx X3 system log to verify that Helium Gateway has started. You can view the log in the web interface or by using logread on the commandline interface (e.g, `logread | grep helium_gateway`). In the logs, you should see messages indicating helium_gateway is starting. If you do not see these, try running the gateway service (`/etc/init.d/helium_gateway server`) or reboot the Cotx X3 as the messages may have rotated out of the log.

*Example logs showing the Helium Gateway service starting*

```
Jul 02 10:35:32 localhost user.notice helium_gateway[238892]: starting server, key: <public key>, version: 1.0.0-alpha.9
Jul 02 10:35:32 localhost user.notice helium_gateway[238892]: starting, module: gateway
Jul 02 10:35:32 localhost user.notice helium_gateway[238892]: starting, module: router
```

Additionally, use the command `cat /var/log/heliumlog/lora-net.log` to view the LoRa Packet Forwarder log. It can provide some clues as to whether the LoRaWAN server configuration is correct. Search the system log for PULL_DATA messages and confirm that they are acknowledged such as `REPORT~ # PULL_DATA sent: 3 (100.00% acknowledged)`. This indicates that the packet forwarder is communicating with the Helium Gateway service running on the Cotx X3 and is getting replies from the service as expected. Note, this is not a foolproof way to confirm it is configured correctly. By default, the LoRaWAN Server settings are configured for the TTN EU server which will also provide acknowledgement. However, the absence of acknowledgement could indicate the the LoRaWAN settings are not correct per the above.

## COMING SOON: Onboard as Data-Only Hotspot

As part of Milestone 1.5 of the Light Gateway project roadmap, you will be able to onboard light gateways such as the Cotx X3 to appear on the coverage map, and then, in Milestone 2, earn HNT for data transfers (but not for Proof of Coverage). See [Light Gateways - BETA](https://docs.helium.com/use-the-network/light-hotspots) for more information. This process will require using the Helium [command-line interface (CLI) Wallet](https://github.com/helium/helium-wallet-rs) to submit the onboarding and assert location transactions.

### CODE REVIEW

**Comment by PharkMillups** [comment]
Awesome. Thanks. Let's see if we can find someone with one of these in hand to get it tested...

File: docs/use-the-network/light-hotspots/guides/Cotx-Networks.mdx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Refer to the [Helium sample LoRa packet forwarder global.conf files](https://github.com/helium/sx1302_hal/tree/helium/hotspot/packet_forwarder) for your region as a guide to the values to use. You can check the values used by the pre-defined frequency plans by viewing the global_conf.json file: `cat /root/helium/lora_net/lora_pkt_fwd/global_conf.json`
+
+## Install Helium Gateway (gateway-rs) on the Cotx X3
+
+If your supported LoRa gateway did not come with helium-gateway pre-installed, manual installation requires you to:
+
+1. Configure the packet forwarder on the gateway to forward to the helium-gateway application. This varies per gateway but the goal is to set the packet forwarder to forward to the (default) configured helium-gateway on `127.0.0.1` at udp port `1680`
+2. Connect to the Cotx X3 over SSH. 
+3. `scp` a downloaded release package for the supported platform to the gateway. Copy the Helium Gateway package to the Cotx X3. Run this command from on the computer within the directory containing the package `scp helium-gateway-<version>-<platform>.deb <gateway>:/tmp/`.
+4. Install the Helium Gateway package `dpkg --install /tmp/helium-gateway-<version>-<platform>.deb`.
```
Comment (madninja) [inline docs/use-the-network/light-hotspots/guides/Cotx-Networks.mdx]: you are missing the actual platform name in there. Which means you will need to issue a PR against gateway-rs to create a package for it.

**Review by madninja — CHANGES_REQUESTED** [review]

File: docs/use-the-network/light-hotspots/guides/Cotx-Networks.mdx (hunk 1 above)
Comment (cotxliang) [reply docs/use-the-network/light-hotspots/guides/Cotx-Networks.mdx]: > you are missing the actual platform name in there. Which means you will need to issue a PR against gateway-rs to create a package for it.

Ok, Thanks! I have successfully added the missing platform name. Is there anything else that needs to be changed?

File: docs/use-the-network/light-hotspots/guides/Cotx-Networks.mdx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+2. Connect to the Cotx X3 over SSH. 
+3. Copy the Helium Gateway package to the Cotx X3. The latest [source code](https://github.com/helium/gateway-rs) and pre-built [releases](https://github.com/helium/gateway-rs/releases) are available on GitHub. Option 1: Download from GitHub: On the Cotx X3 run `cd /tmp` and `wget -O helium-gateway-<version>-cotx.deb https://github.com/helium/gateway-rs/releases/download/<version>/helium-gateway-<version>-cotx.deb`. Option 2: Copy from another computer: Run this command from on the computer within the directory containing the package `scp helium-gateway-<version>-cotx.deb /tmp/`.
+4. Install the Helium Gateway package `dpkg --install /tmp/helium-gateway-<version>-cotx.deb`.
+5. The default region of the gateway is `US915`, if your region is different you can set the right one in `/etc/helium_gateway/settings.toml`. Just add the following line : `region = "<region>"` . Possible values are : `US915| EU868 | EU433 | CN470 | CN779 | AU915 | AS923 | KR920 | IN865`.
```
Comment (madninja) [inline docs/use-the-network/light-hotspots/guides/Cotx-Networks.mdx]:
```suggestion
5. The default region of the gateway is `US915`, if your region is different you can set the right one in `/etc/helium_gateway/settings.toml`. Just add the following line : `region = "<region>"` . Possible values are : `US915 | EU868 | EU433 | CN470 | CN779 | AU915 | AS923_1 | AS923_2 | AS923_3 | AS923_4 | KR920 | IN865`.
```

**Comment by madninja** [comment]
Almost there! This guide is not linked anywhere. Can you add it to the 'use-the-network/light-hotspots.mdx` page near the bottom with the list of other guides? Make it similar to the Dragino entry

**Comment by cotxliang** [comment]
> Almost there! This guide is not linked anywhere. Can you add it to the 'use-the-network/light-hotspots.mdx` page near the bottom with the list of other guides? Make it similar to the Dragino entry

emmm. I found that the link between dragino and kerlink in light-hotspots.mdx doesn't seem right.

File: docs/use-the-network/light-hotspots/light-hotspots.mdx [hunk 3]
```diff
@@ -172,6 +172,8 @@ All Hotspots will not need to follow the Blockchain.
 - [Kerlink gateways](/mine-hnt/light-hotspots/guides/kerlink)
   - Kerlink gateways will only mine HNT as Data Only Hotspot unless it is
     specified as a Helium-Compatible Kerlink Miner.
+- [Cotx gateways](/docs/use-the-network/light-hotspots/guides/cotx.mdx)
```
Comment (madninja) [inline docs/use-the-network/light-hotspots/light-hotspots.mdx]:
```suggestion
- [Cotx gateways](/use-the-network/light-hotspots/guides/cotx)
```

I think path and extension need to be removed. You can also move it to the same folder as the Dragino guide btw

Comment (cotxliang) [reply docs/use-the-network/light-hotspots/light-hotspots.mdx]: > I think path and extension need to be removed. You can also move it to the same folder as the Dragino guide btw

OK, the question has been modified.

**Review by madninja — APPROVED** [review]

**Pull request merged**